### PR TITLE
DBZ-941 Add "snapshot" snapshot locking mode

### PR DESCRIPTION
**Pull request by grzegorz8**
The entire initial snapshot process is executed in snapshot transaction
isolation level.

### CODE REVIEW

**Comment by grzegorz8** [comment]
Depends on https://github.com/debezium/debezium/pull/660

**Comment by grzegorz8** [comment]
@jpechane I would appreciate code review :)

File: debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerSnapshotChangeEventSource.java [hunk 1]
```diff
@@ -82,22 +97,26 @@ protected void lockTablesForSchemaSnapshot(ChangeEventSourceContext sourceContex
             jdbcConnection.connection().setTransactionIsolation(Connection.TRANSACTION_REPEATABLE_READ);
             ((SqlServerSnapshotContext)snapshotContext).preSchemaSnapshotSavepoint = jdbcConnection.connection().setSavepoint("dbz_schema_snapshot");
             LOGGER.info("Schema locking was disabled in connector configuration");
-            return;
-        }
-        LOGGER.info("Executing schema locking");
+        } else if (connectorConfig.getSnapshotLockingMode() == SnapshotLockingMode.EXCLUSIVE) {
```
Comment (jpechane) [inline debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerSnapshotChangeEventSource.java]: Please use formatting
```
}
else if (...
```

File: debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerSnapshotChangeEventSource.java [hunk 2]
```diff
@@ -82,22 +97,26 @@ protected void lockTablesForSchemaSnapshot(ChangeEventSourceContext sourceContex
...
+                    }
 
-                LOGGER.info("Locking table {}", tableId);
+                    LOGGER.info("Locking table {}", tableId);
 
-                statement.executeQuery("SELECT * FROM " + tableId.table() + " WITH (TABLOCKX)").close();
+                    statement.executeQuery("SELECT * FROM " + tableId.table() + " WITH (TABLOCKX)").close();
+                }
             }
+        } else if (connectorConfig.getSnapshotLockingMode() == SnapshotLockingMode.SNAPSHOT) {
```
Comment (jpechane) [inline debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerSnapshotChangeEventSource.java]: `else if` to a seprate line please

File: debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerSnapshotChangeEventSource.java [hunk 3]
```diff
@@ -82,22 +97,26 @@ protected void lockTablesForSchemaSnapshot(ChangeEventSourceContext sourceContex
...
+            ((SqlServerSnapshotContext)snapshotContext).preSchemaSnapshotSavepoint = jdbcConnection.connection().setSavepoint("dbz_schema_snapshot");
+        } else {
```
Comment (jpechane) [inline debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerSnapshotChangeEventSource.java]: ame as above

File: debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerSnapshotChangeEventSource.java (hunk 1 above)
Comment (grzegorz8) [reply debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerSnapshotChangeEventSource.java]: done

**Comment by jpechane** [comment]
@grzegorz8 Applied, thanks! Could you please add a docs PR too?

**Comment by grzegorz8** [comment]
@jpechane Sure, I will.

**Pull request merged**